### ISSUE TITLE
Get LMS groups from the LMS app's requestGroups method

### ISSUE THREAD

**Issue by seanh** [hypothesis/client#1908]
Part of the LMS sections feature: https://github.com/hypothesis/lms/milestone/10

Depends on https://github.com/hypothesis/lms/issues/1555

Change the client to retrieve the list of groups from the LMS app's new `'requestGroups'` method (https://github.com/hypothesis/lms/issues/1555), instead of getting them from the config returned by the `'requestConfig'` method.

## Background

See <https://github.com/hypothesis/lms/issues/1555> for background information on postMessage-JSON-RPC communications between the Hypothesis client and the LMS app (under _Background on postMessage-JSON-RPC_).

See <https://github.com/hypothesis/lms/milestone/10> for background on the Canvas sections feature.

## Problem

In the future the LMS app is going to have to make requests to Canvas's sections API before it can generate the Hypothesis groups for an assignment. Specifically, the LMS frontend is going to have to [call a new LMS backend sync API](https://github.com/hypothesis/lms/milestone/14) and receive the groups from this sync API's response. This would lead to a significant slow down in the assignment launch user experience for all LMS users, if the LMS app had to wait until it had this sync API response before it could inject the Via iframe and begin loading Via and the client.

We believe this will make all assignment launches **about 1.3 seconds slower** than they already are (and they're pretty slow already). It's the difference between sequences 1 and 4 in this sequence diagram:

![sections_times](https://user-images.githubusercontent.com/22498/76650148-e5caf680-6559-11ea-851c-5af275368d20.png)

## Solution

We're going to:

1. **Separate issue:** add a new `'requestGroups'` method to the LMS app's JSON-RPC server, that returns the list of groups. That's this ticket: https://github.com/hypothesis/lms/issues/1555

2. **This issue:** change the client to send _two_ JSON-RPC requests to the LMS app on boot up, not one:

   1. The `'requestConfig'` request that the client already sends
   2. A new `'requestGroups'` request

   The `'requestGroups'` request can be sent at the same time as the `'requestConfig'` request, but the `'requestGroups'` _response_ may not come back as quickly because the LMS app has to wait for its sync API response before it can send the `'requestGroups'` response to the client. The client:

   1. Must retrieve the groups from the `'requestGroups'` response, not from the `'requestConfig'` response. The client mustn't read the groups from the `'requestConfig'` response at all anymore
   2. Must be able to wait a long time for the `'requestGroups'` response, and then continue loading once it finally does receive the groups
   3. Should be able to do as much of its loading as possible using only the `'requestConfig'` response, without the groups

3. **Separate issue:** remove the groups from the config returned by the LMS app's `'requestConfig'` method, now that the client no longer retrieves the groups from this config. That's this issue: https://github.com/hypothesis/lms/issues/1556

Related to https://github.com/hypothesis/client/issues/1793

**Comment by LMS007** [comment]
Last bit required by this ticket. In LMS we need to change the initial value from the groups filter to `rpc:requestGroups` to let the client know the fetch them via RPC.

**Comment by LMS007** [comment]
Blocked by hypothesis/lms#1556 as the value will trigger the client to use RPC to fetch the groups.

**Comment by LMS007** [comment]
Closing this issue as the remaining work is to be done on LMS.
